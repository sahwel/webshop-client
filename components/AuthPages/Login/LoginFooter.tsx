import Link from 'next/link';
import Button from '../../common/Form/Button/Button';

const LoginFooter = () => {
  return (
    <div>
      <Button children="Sign in" type="submit" />
      <Link href="/register" passHref>
        <a className="text-center mt-2 cursor-pointer block">
          Donâ€™t have an account? <span className="font-bold">Register</span>
        </a>
      </Link>
    </div>
  );
};

export default LoginFooter;
