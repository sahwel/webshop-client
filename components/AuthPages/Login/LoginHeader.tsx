import React from 'react';

const LoginHeader = () => {
  return (
    <div className="select-none">
      <h1 className="text-[300%]">Let’s sign you in!</h1>
      <h3>Welcome back,</h3>
      <h3>you’ve been missed!</h3>
    </div>
  );
};

export default LoginHeader;
