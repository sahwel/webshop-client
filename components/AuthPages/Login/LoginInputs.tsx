import {
  faEnvelope,
  faEyeSlash,
  faEye,
} from '@fortawesome/free-regular-svg-icons';
import Link from 'next/link';
import { useCallback, useState } from 'react';
import Input from '../../common/Form/Input/Input';
import InputContainer from '../../common/Form/InputContainer';

const LoginInputs = ({}) => {
  const [isPasswordShow, setIsPasswordShow] = useState(false);
  const handleIsPasswordShowChange = useCallback(() => {
    setIsPasswordShow((oldState) => !oldState);
  }, []);

  return (
    <InputContainer>
      <Input type="text" placeholder="Email" icon={faEnvelope} />
      <div>
        <Input
          type={isPasswordShow ? 'text' : 'password'}
          placeholder="Password"
          icon={isPasswordShow ? faEye : faEyeSlash}
          iconClassName="cursor-pointer"
          onIconClick={handleIsPasswordShowChange}
        />
        <Link href="/forgot">
          <p className="text-right text-sm cursor-pointer mt-1">
            Forgot you password?
          </p>
        </Link>
      </div>
    </InputContainer>
  );
};

export default LoginInputs;
