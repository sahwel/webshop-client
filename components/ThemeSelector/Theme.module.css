
@tailwind components;

@layer components {
    .theme {
       @apply rounded-full p-2  flex justify-center items-center
    }
}