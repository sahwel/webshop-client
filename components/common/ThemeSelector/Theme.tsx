import { FunctionComponent, ReactNode } from 'react';
import cl from 'classnames';
import styles from './theme.module.css';

interface ThemeProps {
  children?: ReactNode;
  isHighlighted: boolean;
  highlightedClass: string;
}

const Theme: FunctionComponent<ThemeProps> = ({
  children,
  isHighlighted,
  highlightedClass,
}) => {
  return (
    <div className={cl({ [highlightedClass]: isHighlighted }, styles.theme)}>
      {children}
    </div>
  );
};

export default Theme;
