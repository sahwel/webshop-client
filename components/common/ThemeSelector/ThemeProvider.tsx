import {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useLayoutEffect,
  useMemo,
  useState,
} from 'react';
import { ThemeContext, ThemeContextModel } from './ThemeContext';
import { Themes } from './definitions';

interface ThemeProviderProps {
  children?: ReactNode;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({ children }) => {
  const [currentTheme, setCurrentTheme] = useState<Themes>('light');

  useLayoutEffect(() => {
    // todo: get alternative solution
    const theme = localStorage.theme ? localStorage.theme : 'light';

    if (theme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    setCurrentTheme(theme);

    return () => {};
  }, []);
  const contextValue: ThemeContextModel = useMemo(
    () => ({
      setTheme() {
        setCurrentTheme((oldTheme) => {
          const newTheme = oldTheme === 'light' ? 'dark' : 'light';

          if (newTheme === 'dark') {
            document.documentElement.classList.add('dark');
          } else {
            document.documentElement.classList.remove('dark');
          }
          localStorage.theme = newTheme;
          return newTheme;
        });
      },
      getTheme() {
        return currentTheme;
      },
    }),
    [currentTheme]
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;

export function useTheme() {
  const value = useContext(ThemeContext);
  return value;
}
