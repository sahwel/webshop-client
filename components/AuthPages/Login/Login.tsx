import { useCallback, useState } from 'react';
import LoginFooter from './LoginFooter';
import LoginHeader from './LoginHeader';
import LoginInputs from './LoginInputs';

const Login = () => {
  return (
    <div>
      <form className="pt-20 grid gap-16 ">
        <LoginHeader />
        <LoginInputs />
        <LoginFooter />
      </form>
    </div>
  );
};

export default Login;
