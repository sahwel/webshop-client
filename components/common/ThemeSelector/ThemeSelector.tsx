import { SunIcon } from '@heroicons/react/outline';
import { MoonIcon } from '@heroicons/react/solid';
import { useCallback, useLayoutEffect, useState } from 'react';
import cl from 'classnames';
import Theme from './Theme';
import styles from './ThemeSelector.module.css';
import { useTheme } from './ThemeProvider';

const ThemeSelector = () => {
  const siteTheme = useTheme();
  const handleChange = useCallback(() => {
    siteTheme.setTheme();
  }, []);

  return (
    <div className={styles.theme_sececltor} onClick={handleChange}>
      <Theme
        highlightedClass={styles.theme_sun_highlited}
        isHighlighted={siteTheme.getTheme() === 'light'}
      >
        <SunIcon className={cl('icon-6', styles.theme_sun)} />
      </Theme>

      <Theme
        highlightedClass={styles.theme_moon_highlited}
        isHighlighted={siteTheme.getTheme() === 'dark'}
      >
        <MoonIcon
          className={cl('icon-6', styles.theme_moon, {
            ['!text-white ']: siteTheme.getTheme() === 'dark',
          })}
        />
      </Theme>
    </div>
  );
};

export default ThemeSelector;
