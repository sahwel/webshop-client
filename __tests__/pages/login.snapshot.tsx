import { render, renderHook } from '@testing-library/react';
import Login from '@/pages/login';
import {
  mockUseThemeSelector,
  themeSelectorWrapper,
} from '../../testUtils/MockThemeSelector';
import { useTheme } from '@/components/common/ThemeSelector/ThemeProvider';
import React from 'react';

React.useContext = mockUseThemeSelector;

describe('login snapshot', () => {
  let _result;
  beforeEach(() => {
    const { result } = renderHook(() => useTheme(), {
      wrapper: themeSelectorWrapper,
    });

    _result = result;
  });
  it('renders homepage unchanged', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
});
