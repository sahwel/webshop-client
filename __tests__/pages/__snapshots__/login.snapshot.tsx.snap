// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot renders homepage unchanged 1`] = `
<div>
  <div
    class="theme_sececltor"
  >
    <div
      class="theme_sun_highlited theme"
    >
      <svg
        aria-hidden="true"
        class="icon-6 theme_sun"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="theme"
    >
      <svg
        aria-hidden="true"
        class="icon-6 theme_moon"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
        />
      </svg>
    </div>
  </div>
  <div
    class="auth_pages_container"
  >
    <div
      class="blob_container"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="blob"
          class="blob"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
    </div>
    <form
      class="pt-20 grid gap-16 "
    >
      <div
        class="select-none"
      >
        <h1
          class="text-[300%]"
        >
          Let’s sign you in!
        </h1>
        <h3>
          Welcome back,
        </h3>
        <h3>
          you’ve been missed!
        </h3>
      </div>
      <div
        class="input_container"
      >
        <div
          class="input_container"
        >
          <input
            class="input"
            placeholder="Email"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope input_icon icon-7"
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.65 28.65 64 64 64H448C483.3 64 512 92.65 512 128V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V128zM48 128V150.1L220.5 291.7C241.1 308.7 270.9 308.7 291.5 291.7L464 150.1V127.1C464 119.2 456.8 111.1 448 111.1H64C55.16 111.1 48 119.2 48 127.1L48 128zM48 212.2V384C48 392.8 55.16 400 64 400H448C456.8 400 464 392.8 464 384V212.2L322 328.8C283.6 360.3 228.4 360.3 189.1 328.8L48 212.2z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div>
          <div
            class="input_container"
          >
            <input
              class="input"
              placeholder="Password"
              type="password"
            />
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-eye-slash input_icon cursor-pointer icon-7"
              data-icon="eye-slash"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M150.7 92.77C195 58.27 251.8 32 320 32C400.8 32 465.5 68.84 512.6 112.6C559.4 156 590.7 207.1 605.5 243.7C608.8 251.6 608.8 260.4 605.5 268.3C592.1 300.6 565.2 346.1 525.6 386.7L630.8 469.1C641.2 477.3 643.1 492.4 634.9 502.8C626.7 513.2 611.6 515.1 601.2 506.9L9.196 42.89C-1.236 34.71-3.065 19.63 5.112 9.196C13.29-1.236 28.37-3.065 38.81 5.112L150.7 92.77zM189.8 123.5L235.8 159.5C258.3 139.9 287.8 128 320 128C390.7 128 448 185.3 448 256C448 277.2 442.9 297.1 433.8 314.7L487.6 356.9C521.1 322.8 545.9 283.1 558.6 256C544.1 225.1 518.4 183.5 479.9 147.7C438.8 109.6 385.2 79.1 320 79.1C269.5 79.1 225.1 97.73 189.8 123.5L189.8 123.5zM394.9 284.2C398.2 275.4 400 265.9 400 255.1C400 211.8 364.2 175.1 320 175.1C319.3 175.1 318.7 176 317.1 176C319.3 181.1 320 186.5 320 191.1C320 202.2 317.6 211.8 313.4 220.3L394.9 284.2zM404.3 414.5L446.2 447.5C409.9 467.1 367.8 480 320 480C239.2 480 174.5 443.2 127.4 399.4C80.62 355.1 49.34 304 34.46 268.3C31.18 260.4 31.18 251.6 34.46 243.7C44 220.8 60.29 191.2 83.09 161.5L120.8 191.2C102.1 214.5 89.76 237.6 81.45 255.1C95.02 286 121.6 328.5 160.1 364.3C201.2 402.4 254.8 432 320 432C350.7 432 378.8 425.4 404.3 414.5H404.3zM192 255.1C192 253.1 192.1 250.3 192.3 247.5L248.4 291.7C258.9 312.8 278.5 328.6 302 333.1L358.2 378.2C346.1 381.1 333.3 384 319.1 384C249.3 384 191.1 326.7 191.1 255.1H192z"
                fill="currentColor"
              />
            </svg>
          </div>
          <a
            class="text-right text-sm block cursor-pointer mt-1"
            href="/forgot"
          >
            Forgot your password?
          </a>
        </div>
      </div>
      <div>
        <button
          class="button"
          type="submit"
        >
          Sign in
        </button>
        <a
          class="text-center mt-2 cursor-pointer block"
          href="/register"
        >
          Don’t have an account? 
          <span
            class="font-bold"
          >
            Register
          </span>
        </a>
      </div>
    </form>
  </div>
</div>
`;
